spec/rubocop/cop/style/map_to_hash_spec.rb

```diff
@@ -37,7 +37,7 @@
           RUBY
 
           expect_correction(<<~RUBY)
-            foo&.to_h { |x| [x, x * 2] }
+            foo.to_h { |x| [x, x * 2] }
```

Comment:
This is fixed incorrectly.
```console
2.7.6 :018 > foo = [1, 2, 3, 4]
 => [1, 2, 3, 4] 
2.7.6 :019 > foo&.map { |x| [x, x * 2] }&.to_h
 => {1=>2, 2=>4, 3=>6, 4=>8} 
2.7.6 :020 > foo.to_h { |x| [x, x * 2] }
 => {1=>2, 2=>4, 3=>6, 4=>8} 
```
```console
2.7.6 :021 > foo = nil
 => nil 
2.7.6 :022 > foo&.map { |x| [x, x * 2] }&.to_h
 => nil 
2.7.6 :023 > foo.to_h { |x| [x, x * 2] }
 => {} 
```

Reply:
Ah, I overlooked the case where behavior changes when `nil`. For example, there is no way to know if argument value is `nil`. This should retain the safe navigation operator.